tests/unit/IncorporationDateTime.spec.ts

```diff
@@ -0,0 +1,128 @@
+    expect(radioIsFutureEffective.attributes('aria-checked')).toBe('false')
+  })
+
+  it('confirms the selector fields are disabled if future effective is NOT selected', () => {
+    const wrapper = wrapperFactory({ incorporationDateTime: dateTimeDefault })
+
+    expect(wrapper.find('#date-text-field').attributes('disabled')).toBe('disabled')
+    expect(wrapper.find('#hour-selector').attributes('disabled')).toBe('disabled')
+    expect(wrapper.find('#minute-selector').attributes('disabled')).toBe('disabled')
+    expect(wrapper.find('#am-pm-selector').attributes('disabled')).toBe('disabled')
+  })
+
+  it('confirms the selector fields are NOT disabled if future effective is selected', async () => {
+    const wrapper = wrapperFactory({ incorporationDateTime: dateTimeDefault })
+
+    const radioInput = wrapper.findAll('input[type="radio"]')
+    const radioIsFutureEffective = radioInput.at(1)
+
+    await radioIsFutureEffective.trigger('click')
+
+    expect(wrapper.find('#date-text-field').attributes('disabled')).toBe(undefined)
+    expect(wrapper.find('#hour-selector').attributes('disabled')).toBe(undefined)
+    expect(wrapper.find('#minute-selector').attributes('disabled')).toBe(undefined)
+    expect(wrapper.find('#am-pm-selector').attributes('disabled')).toBe(undefined)
+  })
+
+  it('confirms the selector fields are toggled to disabled if Immediate Filing is selected', async () => {
+    const wrapper = wrapperFactory({ incorporationDateTime: dateTimeDefault })
+
+    const radioInput = wrapper.findAll('input[type="radio"]')
+    const radioIsImmediate = radioInput.at(0)
+    const radioIsFutureEffective = radioInput.at(1)
+
+    await radioIsFutureEffective.trigger('click')
+
+    expect(wrapper.find('#date-text-field').attributes('disabled')).toBe(undefined)
+    expect(wrapper.find('#hour-selector').attributes('disabled')).toBe(undefined)
+    expect(wrapper.find('#minute-selector').attributes('disabled')).toBe(undefined)
+    expect(wrapper.find('#am-pm-selector').attributes('disabled')).toBe(undefined)
+
+    await radioIsImmediate.trigger('click')
+
+    expect(wrapper.find('#date-text-field').attributes('disabled')).toBe('disabled')
+    expect(wrapper.find('#hour-selector').attributes('disabled')).toBe('disabled')
+    expect(wrapper.find('#minute-selector').attributes('disabled')).toBe('disabled')
+    expect(wrapper.find('#am-pm-selector').attributes('disabled')).toBe('disabled')
+  })
+
+  it('emits a valid state when the Immediate Filing is selected', async () => {
+    const wrapper = wrapperFactory({ incorporationDateTime: dateTimeDefault })
+
+    const radioInput = wrapper.findAll('input[type="radio"]')
+    const radioIsImmediate = radioInput.at(0)
+    await radioIsImmediate.trigger('click')
+
+    // Verify the Valid emit event is true
+    expect(wrapper.emitted().valid).toEqual([[true]])
+  })
+
+  it('emits an invalid state when the Future Effective is selected and no date is selected', async () => {
```

Comment:
Add test for checking valid Future Effective state.

Reply:
for sure!

Reply:
added!